File: app-java/src/test/java/tech/thdev/java_udemy_sample/NullSafetyTest.java
package tech.thdev.java_udemy_sample;

import android.support.annotation.Nullable;

import org.junit.Test;

/**
 * Created by tae-hwan on 10/6/16.
 *
 * 실행 방법
 *
 * @Test가 포함된 함수의 오른쪽 마우스를 누른 상태에서 Run을 하시면 됩니다
 */

public class NullSafetyTest {

    /**
     * 일반적인 null 처리 방법을 소개합니다
     * @throws InterruptedException
     */
    @Test
    public void testNullCheck() throws InterruptedException {
        String name = null;
        System.out.print(getLength(name));
    }

    private int getLength(@Nullable String name) {
        if (name != null) {
            return name.length();
        }
        return 0;
    }
}

File: app-kotlin/src/test/java/tech/thdev/kotlin_udemy_sample/NullSafetyTest.kt
package tech.thdev.kotlin_udemy_sample

import org.junit.Test

/**
 * Created by tae-hwan on 10/6/16.
 */
class NullSafetyTest {

    @Test
    @Throws(InterruptedException::class)
    fun testNullCheck() {
        val name: String? = null
        println(getLength(name))
    }

    private fun getLength(name: String?): Int {
        if (name != null) {
            return name.length
        }
        return 0
    }





























    private fun getLength2(name: String?): Int? {
        return name?.length
    }

    private fun getLength3(name: String?): Int {
        return try {
            name!!.length
        } catch (e: Exception) {
            0
        }
    }
}